spec/c14n_spec.rb

```diff
@@ -164,6 +164,10 @@
             canonicalize!(allow_query: :all).
             to_s.should eql('http://example.com/path?%ED=view')
         end
+
+        it 'does not error when there are bad things in query keys when allow_query isn\'t :all' do
+          BLURI('http://some.com/a/path?%E2').canonicalize!
```

Comment:
Should we also have some examples of using `canonicalize!` with the sorts of parameters that:

1. Bouncer/Transition will use on canonicalization via significant query string params
2. A dev debugging an issue might use from `irb` – for example a demonstration that `BLURI('http://example.com/some?significant=1&query_params=2').canonicalize!(:significant)` and `BLURI('http://example.com/some?significant=1&query_params=2').canonicalize!('significant')` behave in the same way